src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -866,7 +866,7 @@ public static Integer typeScale(String typeName) {
             case "decimal":
                 return 34;
         }
```

Comment:
Since we can no longer return null here, I think we might as well return `int` instead of `Integer`. Saves us some GC.

Reply:
Sometime, null is actually a valid value. It will handled correctly when SQL-710 is implemented.
I'm leaving the return type as an Object for now, just so that we won't forget that it can be null.
But I update the method callings these helpers to use bsonint32 (which can handle a null value) instead of new BsonInt32 (which can't handle a null value).